modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java

```diff
@@ -304,7 +305,7 @@ public RecordBinaryViewImpl(InternalTable tbl, SchemaRegistry schemaReg, TableMa
         HashSet<BinaryRow> keys = new HashSet<>(recs.size());
 
         for (Tuple keyRec : recs) {
-            final Row keyRow = marshaller().marshal(keyRec);
+            final Row keyRow = marshal(keyRec, false);
```

Comment:
Are you sure that should be false? the method name is 'deleteAllExactAsync' so we should check not only key